src/nrniv/neuronapi.cpp

```diff
@@ -158,159 +168,277 @@ void nrn_segment_diam_set(Section* const sec, const double x, const double diam)
     }
 }
 
-double nrn_rangevar_get(Symbol* sym, Section* sec, double x) {
+double nrn_rangevar_get(Section* sec, double x, const char* var_name) {
+    Symbol* sym = hoc_table_lookup(var_name, hoc_built_in_symlist);
     return *nrn_rangepointer(sec, sym, x);
 }
 
-void nrn_rangevar_set(Symbol* sym, Section* sec, double x, double value) {
+void nrn_rangevar_set(Section* sec, double x, const char* var_name, double value) {
+    Symbol* sym = hoc_table_lookup(var_name, hoc_built_in_symlist);
     *nrn_rangepointer(sec, sym, x) = value;
 }
 
-void nrn_rangevar_push(Symbol* sym, Section* sec, double x) {
-    hoc_push(nrn_rangepointer(sec, sym, x));
+RangeVar nrn_rangevar_new(Section* sec, double x, const char* var_name) {
```

Comment:
No idea if this would have worked before, but if we're embedding the `node` inside the `RangeVar` object, wouldn't that mean we might be asking for trouble if `nseg` changes?

Reply:
I am not sure I can answer that. True that this code assumes RangeVar is tied to a node/segment. If that must not happen we should change this for do the lookup on the very last moment @nrnhines